modules/genomic_browser/php/endpoints/filemanager.class.inc

```diff
@@ -0,0 +1,352 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used for populating the Genomic Browser.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Genomic_Browser
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\genomic_browser\endpoints;
+
+use LORIS\genomic_browser\Provisioners\FilesProvisioner;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Api\Endpoint;
+use \LORIS\Middleware\ETagCalculator;
+
+require_once 'uploading/genomicfile.class.inc';
+
+/**
+ * The NDB_Filter_Form_GenomicBrowser Class.
+ *
+ * This class provides the php code to create the menu filter form that
+ * allows filemanager to be processed as part of the genomic browser.
+ *
+ * @category Loris
+ * @package  Genomic_Browser
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class FileManager extends Endpoint implements ETagCalculator
+{
+    /**
+     * A cache of the endpoint results, so that it doesn't need to be
+     * recalculated for the ETag and handler.
+     */
+    private $_cache;
+
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        case 'POST':
+            return $this->_handlePOST($request);
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values = $request->getQueryParams();
+        // User permission verification.
+        $values = $this->_verifyPermissions($request, $values);
+        if (isset($values['error'])) {
+            switch ($values['error']) {
+            case 'BadRequest':
+                return new \LORIS\Http\Response\BadRequest(
+                    'Failed to retrieve data for create timepoint.'
+                );
+            case 'Forbidden':
+                return new \LORIS\Http\Response\Forbidden(
+                    'Forbidden'
```

Comment:
just fyi you don't need this. The class will return a default message of "forbidden"

Reply:
@johnsaigle I'm confused on how that happens. You're advising I delete the case 'Forbidden' but wouldn't that just skip to default?

Reply:
No I mean you literally don't need to provide the string argument. 

`return new \LORIS\Http\Response\Forbidden();` will do the trick.